src/main/java/org/elasticsearch/index/store/Store.java

```diff
@@ -630,6 +630,7 @@ public MetadataSnapshot(Map<String, StoreFileMetaData> metadata) {
                     // Lucene checks the checksum after it tries to lookup the codec etc.
                     // in that case we might get only IAE or similar exceptions while we are really corrupt...
                     // TODO we should check the checksum in lucene if we hit an exception
+                    logger.warn("checking segment info integrity (reason [{}])", ex.getMessage());
```

Comment:
I would log the exception here, not just the message, it will provide much more details, and its bad if it happens I would imagine.

Also, I would try and explain it better in the logging, for example, tried to do lightweight check, but failed, resorting to full checksum